modules/web-console/src/main/java/org/apache/ignite/console/routes/AgentDownloadRouter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.console.routes;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.ignite.Ignite;
+
+/**
+ * Router to handle REST API to download Web Agent.
+ */
+public class AgentDownloadRouter extends AbstractRouter {
+    /** Buffer size of 30Mb to handle Web Agent ZIP file manipulations. */
+    private static final int BUFFER_SZ = 30 * 1024 * 1024;
+
+    /** */
+    private File pathToWebAgentZip;
+
+    /** */
+    private final String file;
+
+    /**
+     * @param ignite Ignite.
+     */
+    public AgentDownloadRouter(Ignite ignite, String folder, String file) {
+        super(ignite);
+
+        this.file = file;
+        this.pathToWebAgentZip = Paths.get(folder, file + ".zip").toFile();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void initializeCaches() {
+        // No-op so far.
+    }
+
+    /** {@inheritDoc} */
+    @Override public void install(Router router) {
+        router.get("/api/v1/downloads/agent").handler(this::load);
+    }
+
+    /**
+     * @param ctx Context.
+     */
+    private void load(RoutingContext ctx) {
+        User user = checkUser(ctx);
+
+        if (user != null) {
+            try {
+                ZipFile zip = new ZipFile(pathToWebAgentZip);
```

Comment:
Please check that agent archive is present and throw exception with following message `Missing agent zip on server. Please ask webmaster to upload agent zip!`